Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -718,23 +753,57 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
         extraInfoHolder.length = tofInt.getL();
         // extra info from sub-tracks
         if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
-          isStoredITS[track.getRefITS()] = true;
           const auto& itsOrig = recoData.getITSTrack(contributorsGID[GIndex::ITS]);
           extraInfoHolder.itsClusterMap = itsOrig.getPattern();
         }
         if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
-          isStoredTPC[track.getRefTPC()] = true;
           const auto& tpcOrig = recoData.getTPCTrack(contributorsGID[GIndex::TPC]);
+          extraInfoHolder.tpcInnerParam = tpcOrig.getP();
+          extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
+          extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
+          extraInfoHolder.tpcNClsShared = countTPCSharedCl(tpcOrig, tpcClusRefs, tpcClusShMap, tpcClusAcc);
+        }
+        addToTracksTable(tracksCursor, tracksCovCursor, track, -1, src);
+        addToTracksExtraTable(tracksExtraCursor, extraInfoHolder);
+      }
+      if (src == GIndex::Source::TPCTRD && mFillTracksTPC) {
+        const auto& track = tracksTPCTRD[trackIndex.getIndex()];
```

Comment:
I don't see where do you save TPC Nclusters? BTW, with the current TPC tracking one can have >1 cluster on a given padrow. For the calculation of the tpcChi2NCl it is indeed better to use `tpcOrig.getNClusters()`, but as Nclusters stored in the AOD I would use actual number of contributing padrows, like it is calculated in the snippet I sent you before.

Reply:
I thought we agreed on filling number of shared clusters for now, but maybe I misunderstood. Where should I put `found` and `crossed` for now then? The corresponding columns are dynamic (https://github.com/AliceO2Group/AliceO2/blob/dc091dba2e910273d38d0a4dac48963b427f2adf/Framework/Core/include/Framework/AnalysisDataModel.h#L218-L221), so they can't be used in a file.

Reply:
What I wrote before is that `tpcNClsFindable` is not available in run3 reco output with the same definition as in run2, so better to use a placeholder at the moment. For instance, you can use instead tpcNClsCrossedRows in order to define the differences currently present in the AOD.